lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -198,8 +198,16 @@ def _put(
             except MetadataError as e:
                 fail[k] = str(e)
 
-        results = {
-            "metadata": self._get_dataset_metadata(dataset, list(metadata.keys())),
-            "errors": fail,
-        }
-        return jsonify(results)
+        if len(fail) == len(metadata):
```

Comment:
Is this a sufficient test?  Would if the metadata is the same length?

Reply:
Can you elaborate the concern?

`metadata` is a dict of key/value pairs. `fail` is a dict of key/failure pairs. If the lengths are identical than every key failed. The origin of `fail` is an iteration through `metadata.items()` so `fail` can't have *more* keys, only fewer. If it has fewer, than at least one update succeeded and we report a "partial success". If there was no success, then it's a failure.

Reply:
Right, just ignore me.